cspreports/forms.py

```diff
@@ -0,0 +1,16 @@
+from __future__ import unicode_literals
+
+from django.forms import ModelForm
+from django.forms.models import model_to_dict
+
+from cspreports.models import CSPReport
+
+
+class CSPReportForm(ModelForm):
+    class Meta:
+        model = CSPReport
+        fields = '__all__'
+
+    def __init__(self, message, *args, **kwargs):
+        report = self._meta.model.from_message(message)
+        super(CSPReportForm, self).__init__(model_to_dict(report), *args, **kwargs)
```

Comment:
This seems kind of weird. At first raw data are converted to a model, then converted back to data and provided to a form, which validates them a converts them to a model again. Seem like a way too much conversions back and forth.

Reply:
It really doesn't make sense to use both. At the moment, I think it's better and easier to keep the `from_message` and drop the form.